MoonDawgArduino/MoonDawgArduino/MoonDawgArduino.ino

```diff
@@ -1,146 +1,137 @@
+ */
+// Cameron Schwartzberg
+// 2/18/2024, 7:07 PM
+// 240216-212839-uno
+// The main file that is loaded onto the Arduino, this will call all the functions needed to control the robot.
+// Cameron Schwartzberg, Marco Caliendo, Andrew Barnes, Mitchell Wettig, Jared Hillyer
+
+#include <Servo.h>
+#include <Arduino.h>
+
+#include "PinDefinitions.h"
+#include "Movement.h"
+
+constexpr int MAX_ARRAY_SIZE = 10;
+constexpr int MAX_INPUT_LENGTH = 50;
+
+constexpr char MOVEMENT = 'm';
+constexpr char DIGBELT = 'b';
+constexpr char DIGACT = 'g';
+constexpr char DEPOSITAUGER = 'd';
+constexpr char VIBRATOR = 'v';
+
+
+// Class constructors
+DriveMovement driving = DriveMovement();
+DepositMovement depositing = DepositMovement();
+DigMovement digging = DigMovement();
+
+void commandProcessing(char* tokens[]) {
+
+    char cmd = tokens[0][0];
+    int param1 = atoi(tokens[1]);
+    int param2;
+    if (cmd == MOVEMENT || cmd == DIGBELT) {
+      param2 = atoi(tokens[2]);
+    }
+    else {
+      param2 = (int)tokens[2][0];
+    }
+
+
+    switch (cmd) {
+        case MOVEMENT:
+            driving.drive(param1, param2);
+            break;
+        case DIGBELT:
+            if (param1) {
+                digging.digMotor(param2);
+            }
+            else {
+                digging.digMotorStop();
+            }
+            break;
+        case DIGACT:
+            if (param1) {
+                if (param2 == 'r')
+                {
+                    digging.digActuatorForward();
+                }
+                else if (param2 == 'l')
+                {
+                    digging.digActuatorBackward();
+                }                
+            }
+            else {
+                digging.digActuatorStop();
+            }
+            break;
+        case DEPOSITAUGER:
+            if (param1) {
+                if (param2 == 'f') {
+                    depositing.depositMotorForward();
+                }
+                else if (param2 == 'b')
+                {
+                    depositing.depositMotorBackward();
+                }
+                
+            }
+            else {
+                depositing.depositMotorStop();
+            }
+            break;
+        case VIBRATOR:
+            if (param1) {
+                if (param2 == 'v'){
+                    depositing.depositVibrator();
+                }
+            }
+            else {
+                depositing.depositVibratorStop();
+            }
+        default:
+            break;
+    }
+}
+
+
+void setup() {
+  // put your setup code here, to run once:
+    // For the serial communication from RPi to Arduino
+    Serial.begin(9600);
+
+    driving.driveSetup(LEFT_MOTOR, RIGHT_MOTOR);
+    depositing.depositSetup(DEPOSIT_MOTOR, DEPOSIT_VIBRATOR);
+    digging.digSetup(DIGGING_MOTOR, DIGGING_ACTUATOR);
+}
+
+void loop() {
+  // put your main code here, to run repeatedly
+
+    if (Serial.available() > 0) {
```

Comment:
should move this serial read block into its own function, returning the char array and passing it to command processing

Reply:
Let's note this and look at it in the new semester so we can test it still works correctly on the bot.